app/src/main/java/com/github/houseorganizer/houseorganizer/panels/NavBarActivity.java

```diff
@@ -0,0 +1,61 @@
+package com.github.houseorganizer.houseorganizer.panels;
+
+import android.content.Intent;
+
+import androidx.annotation.IdRes;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.android.material.bottomnavigation.BottomNavigationView;
+import com.google.firebase.firestore.DocumentReference;
+
+import java.util.OptionalInt;
+
+public abstract class NavBarActivity extends AppCompatActivity {
+    protected enum CurrentActivity{
+        MAIN("Main Screen", MainScreenActivity.class),
+        CALENDAR("Calendar", CalendarActivity.class),
+        GROCERIES("Groceries", null), /* TODO */
+        TASKS("Tasks", TaskListActivity.class);
+
+        protected final String name;
+        protected final Class<? extends AppCompatActivity> panelActivity;
+
+        CurrentActivity(String name, Class<? extends AppCompatActivity> panelActivity) {
+            this.name = name;
+            this.panelActivity = panelActivity;
+        }
+
+        protected static CurrentActivity activityWithName(String name) {
+            for (CurrentActivity activity : values()) {
+                if (activity.name.equals(name)) {
+                    System.out.println("!!!!!!!!!" + name);
```

Comment:
Why is there a print here, should it be a Log.d ?

Reply:
It was a debugging print, thanks for the catch! I'm deleting it in the next commit